chain/ethereum/src/capabilities.rs

```diff
@@ -15,28 +13,28 @@ pub struct NodeCapabilities {
     pub traces: bool,
 }
 
-// Take all NodeCapabilities fields into account when ordering
-// A NodeCapabilities instance is considered equal or greater than another
-// if all of its fields are equal or greater than the other
-impl Ord for NodeCapabilities {
-    fn cmp(&self, other: &Self) -> Ordering {
-        match (
+/// Two [`NodeCapabilities`] can only be compared if one is the subset of the
+/// other. No [`Ord`] (i.e. total order) implementation is applicable.
+impl PartialOrd for NodeCapabilities {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        product_order([
             self.archive.cmp(&other.archive),
             self.traces.cmp(&other.traces),
-        ) {
-            (Ordering::Greater, Ordering::Greater) => Ordering::Greater,
-            (Ordering::Greater, Ordering::Equal) => Ordering::Greater,
-            (Ordering::Equal, Ordering::Greater) => Ordering::Greater,
-            (Ordering::Equal, Ordering::Equal) => Ordering::Equal,
-            (Ordering::Less, _) => Ordering::Less,
-            (_, Ordering::Less) => Ordering::Less,
-        }
+        ])
```

Comment:
Isn't this just `(&self.archive, &self.traces).cmp((&other.archive, &other.traces))`.

Reply:
No, Rust tuples define lexicographic ordering, which is not what we use for node capabilities comparisons (e.g. all fields have to be `>=` in order for a `NodeCapabilities` to be `>`, but if we just compare them as tuples, the comparison might stop at `archive` and discard `traces`).

Reply:
I see, thanks